File: components/credentials/src/main/java/org/cloudfoundry/credhub/requests/UserSetRequest.java
package org.cloudfoundry.credhub.requests;

import java.util.Objects;

import javax.validation.Valid;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.cloudfoundry.credhub.ErrorMessages;
import org.cloudfoundry.credhub.credential.UserCredentialValue;

public class UserSetRequest extends BaseCredentialSetRequest<UserCredentialValue> {
  @NotNull(message = ErrorMessages.MISSING_VALUE)
  @Valid
  @JsonProperty("value")
  private UserCredentialValue userValue;

  public UserCredentialValue getUserValue() {
    return userValue;
  }

  public void setUserValue(final UserCredentialValue userValue) {
    this.userValue = userValue;
  }

  @Override
  public UserCredentialValue getCredentialValue() {
    return userValue;
  }

  @Override
  public GenerationParameters getGenerationParameters() {
    return null;
  }

  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    final UserSetRequest that = (UserSetRequest) o;
    return Objects.equals(userValue, that.userValue);
  }

  @Override
  public int hashCode() {
    return Objects.hash(userValue);
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/requests/UserSetRequest.kt
package org.cloudfoundry.credhub.requests

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.Objects
import javax.validation.Valid
import javax.validation.constraints.NotNull
import org.cloudfoundry.credhub.ErrorMessages
import org.cloudfoundry.credhub.credential.UserCredentialValue

class UserSetRequest : BaseCredentialSetRequest<UserCredentialValue?>() {
    @NotNull(message = ErrorMessages.MISSING_VALUE)
    @Valid
    @JsonProperty("value")
    var userValue: UserCredentialValue? = null
    override val generationParameters: GenerationParameters?
        get() = null

    override val credentialValue: UserCredentialValue?
        get() {
            return userValue
        }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as UserSetRequest?
        return Objects.equals(userValue, that!!.userValue)
    }

    override fun hashCode(): Int {
        return Objects.hash(userValue)
    }
}
